src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_DeconstructionAssignmentOperator.cs

```diff
@@ -103,6 +106,88 @@ internal sealed partial class LocalRewriter
 
                 return _factory.Sequence(temps.ToImmutableAndFree(), effects.ToImmutableAndFree(), returnValue);
             }
+
+            void reverseAssignmentsToTargetsIfApplicable()
+            {
+                // The list of deconstruction targets is usually small. Prefer a temporary array which is stack-bound for small numbers of elements.
+                var visitedSymbols = TemporaryArray<Symbol>.Empty;
+
+                if (right is BoundConvertedTupleLiteral or BoundConversion { Operand.Kind: BoundKind.TupleLiteral or BoundKind.ConvertedTupleLiteral }
+                    && effects.init.Any()
```

Comment:
>effects.init.Any() [](http://example.com/codeflow?start=23&length=18)

It would be good to capture what we are trying to infer with this check. I assume we are detecting presence of temp assignments that are supposed to precede assignments to the targets. Correct?

Reply:
Yes, I'll try and make that more clear.